src/utils/render-fragments.js

```diff
@@ -153,13 +148,29 @@ export const getGlossaryUsedLink = (slug, allGlossary) => {
 
 export const getGlossaryItemsUsedInRule = slug => {
 	const keys = []
-	Object.keys(glossaryUsages).forEach(key => {
-		glossaryUsages[key].forEach(({ slug: s }) => {
+
+	/**
+	 * Iterate through references in rules
+	 */
+	Object.keys(glossariesInRules).forEach(key => {
+		glossariesInRules[key].forEach(({ slug: s }) => {
 			if (s === slug && !keys.includes(key)) {
 				keys.push(key)
 			}
 		})
 	})
+
+	/**
+	 * Iterate through references in glossary thyself
+	 */
+	Object.keys(glossariesInGlossaries).forEach(key => {
+		glossariesInGlossaries[key].forEach(({ glossarykey }) => {
+			if (!keys.includes(glossarykey)) {
```

Comment:
It's hard to tell (because the code is hard to read), but it looks like this is including everything. Definitely this isn't recursively looking up definitions inside of other definitions.